nemo/collections/asr/parts/submodules/rnnt_greedy_decoding.py

```diff
@@ -598,12 +599,20 @@ def __init__(
             confidence_method_cfg=confidence_method_cfg,
         )
 
+        self.use_cuda_graph_decoder = use_cuda_graph_decoder
+
         # Depending on availability of `blank_as_pad` support
         # switch between more efficient batch decoding technique
         if self.decoder.blank_as_pad:
-            if loop_labels:
+            if loop_labels and use_cuda_graph_decoder:
+                raise ValueError("loop_labels and use_cuda_graph_decoder is unsupported configuration")
+            elif loop_labels:
                 # default (faster) algo: loop over labels
                 self._greedy_decode = self._greedy_decode_blank_as_pad_loop_labels
+            elif use_cuda_graph_decoder:
+                from nemo.collections.asr.parts.submodules.fast_rnnt_greedy_decoding import RNNTGreedyDecodeFast
+
+                self._greedy_decode = RNNTGreedyDecodeFast(max_symbols_per_step, torch.device("cuda"), self)
```

Comment:
So the init variables auto reset at the end of each `__call__` ?

Reply:
Not sure I understand your question, but I will try to respond. 

No they don't. I made a class with a `__call__` method so I could save some values that I need to initialize (like pinned CPU memory and the cuda graph itself) as instance variables, which do not need to be recreated across decoding calls. If they did have to, this would be very bad for throughput.